enterprise/wan-replication/basic-sample/src/main/resources/hazelcast-wan-source.xml

```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+  ~
+  ~ Licensed under the Apache License, Version 2.0 (the "License");
+  ~ you may not use this file except in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~ http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing, software
+  ~ distributed under the License is distributed on an "AS IS" BASIS,
+  ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  ~ See the License for the specific language governing permissions and
+  ~ limitations under the License.
+  -->
+
+<!--
+  The minimum Hazelcast WAN source sample configuration needed for replicating
+  a map with the WAN replication implementation built in Hazelcast Enterprise.
+-->
+<hazelcast xmlns="http://www.hazelcast.com/schema/config"
+           xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+           xsi:schemaLocation="http://www.hazelcast.com/schema/config
+           http://www.hazelcast.com/schema/config/hazelcast-config-4.0.xsd">
+
+    <!--
+      The name of the source cluster. It doesn't matter for WAN replication.
+      It is set in this sample only to make sure the WAN source and target
+      cluster members don't form one cluster.
+    -->
+    <cluster-name>wan-source</cluster-name>
+
+    <!--
+      The Hazelcast Enterprise license key. This needs to be set in order
+      to run the sample.
+    -->
+    <license-key>YOUR_LICENSE_KEY</license-key>
+
+    <map name="wan-replicated-map">
+        <!--
+          The reference to the `wan-replication` configuration to be used for
+          replicating this map. The name should match to the name set in the
+          referenced `wan-replication` configuration element.
+         -->
+        <wan-replication-ref name="wan-sample">
+            <merge-policy-class-name>PassThroughMergePolicy</merge-policy-class-name>
```

Comment:
The merge policy should be optional (and I don't see it in the programmatic config) so maybe you can remove it from here and other occurrences in XML and YAML.

Reply:
Yeah, I tried to remove the merge policy, but it resulted in an invalid YAML config, so I added it back. For now, I'd add it to programmatic too to have really the same config in the three ways. After fixing the YAML structure, I remove this.

Reply:
Solved with this instead:
```
...
wan-replication-ref:
      wan-sample: {}
```
So the merge policy configurations could be removed. Also created https://github.com/hazelcast/hazelcast/issues/16671 to make it more straightforward to configure wan replication reference with defaults.